google/cloud/spanner_dbapi/parse_utils.py

```diff
@@ -180,27 +177,29 @@ def classify_stmt(query):
     :type query: str
     :param query: A SQL query.
 
-    :rtype: str
-    :returns: The query type name.
+    :rtype: ParsedStatement
+    :returns: parsed statement attributes.
```

Comment:
Is this function considered to be a public/external interface?  Is it possible that someone externally is calling it and might be depending on this return value?  In which case changing the function's signature would be a breaking change, and should be treated as such (with a major-version bump and other discussion/communication).

(Tangent, although breaking changes should only be made with care in clients, I enjoy that it's actually possible to do so.  Over here in backend-code-land, we can basically never change any API, only add new features, because our service is hosted/multitenant and customers can't pin to an old backend version if they're not ready for an update yet.)

Reply:
I dont think so but let me try to confirm it

Reply:
I have left this method as it is and mark it as deprecated and created a new function _classify_stmt to be used

Reply:
If this function is deprecated, you might consider updating its code to issue a `DeprecationWarning`:

https://docs.python.org/3/library/warnings.html#warnings.warn
https://docs.python.org/3/library/exceptions.html#DeprecationWarning

For future reference (not really applicable to this PR):  If a project commonly deprecates functions (which I don't think this project does yet?, but it might as of some future PR), that project may want to write for itself a little utility decorator `@deprecated` that issues the above warning and then calls the base function.  For both convenience and readability.  Projects that prefer third-party dependencies can get such a decorator via the [`Deprecated` package](https://pypi.org/project/Deprecated/).  Python has a [PEP to make this a builtin](https://peps.python.org/pep-0702/) as of Python 3.13, in probably about a year.

Reply:
Please see line 182, I have already done that or am I missing anything

Reply:
Oh oops, I was looking at the commit view, not the overall PR view.  Sorry!  LGTM.